beacon_node/http_api/src/sync_committees.rs

```diff
@@ -108,11 +108,13 @@ fn duties_from_state_load<T: BeaconChainTypes>(
         // have to transition the head to start of the current period).
         //
         // We also need to ensure that the load slot is after the Altair fork.
+        let anchor_info = chain.store.get_anchor_info();
+        let state_upper_limit = anchor_info.state_upper_limit;
```

Comment:
We should account for the case where `state_upper_limit` is equal to `STATE_UPPER_LIMIT_NO_RETAIN` which we use to represent the configuration where _no_ historic states are stored. The "no retain" value is very high (`2^64 - 1`), so if we naively `max` it in here then the `load_slot` will always be `2^64 - 1`, which won't work.

Reply:
done , do check it out once free!